server/gfmd/db_journal_apply.c

```diff
@@ -946,6 +946,63 @@ db_journal_apply_mdhost_remove(gfarm_uint64_t seqnum, char *name)
 	return (e);
 }
 
+/**********************************************************/
+/* user_auth */
+
+static gfarm_error_t
+db_journal_apply_user_auth_add(gfarm_uint64_t seqnum,
+	struct db_user_auth_arg *arg)
+{
+	gfarm_error_t e;
+	struct user *u;
+	struct user_auth *ua;
+
+	u = user_tenant_lookup(arg->username);
+	if (u == NULL)
+		return (GFARM_ERR_NO_SUCH_USER);
+
+	e = user_enter_auth_id(u, arg->auth_method, arg->auth_user_id, 0, &ua);
+	if (e != GFARM_ERR_NO_ERROR)
+		return (e);
+
+	user_set_user_auth_metadata_in_cache(ua,
+		arg->auth_method, arg->auth_user_id);
+	return (GFARM_ERR_NO_ERROR);
+}
+
+static gfarm_error_t
+db_journal_apply_user_auth_modify(gfarm_uint64_t seqnum,
+	struct db_user_auth_arg *arg)
+{
+	struct user *u;
+	struct user_auth *ua;
+
+	u = user_tenant_lookup(arg->username);
+	if (u == NULL)
+		return (GFARM_ERR_NO_SUCH_USER);
+
+	ua = user_lookup_auth_id(u, arg->auth_method, arg->auth_user_id);
+	if (ua == NULL)
+		return (GFARM_ERR_NO_SUCH_OBJECT);
+
+	user_set_user_auth_metadata_in_cache(ua,
+		arg->auth_method, arg->auth_user_id);
+	return (GFARM_ERR_NO_ERROR);
+}
+
+static gfarm_error_t
+db_journal_apply_user_auth_remove(gfarm_uint64_t seqnum,
+	struct db_user_auth_remove_arg *arg)
+{
+	struct user *u;
+
+	u = user_tenant_lookup(arg->username);
+	if (u == NULL)
+		return (GFARM_ERR_NO_SUCH_USER);
+
+	return (user_remove_auth_id(u, arg->auth_method));
```

Comment:
change this function name from user_remove_auth_id() to user_auth_id_remove(),
because it is more convenient to grep by "user_auth"